src/strings_internal.h

```diff
@@ -122,19 +132,28 @@ class StringBuilder {
 	}
 
 	/**
-	 * Add a string using the strecpy/strecat-esque calling signature.
-	 * @param function The function to pass the current and last location to,
-	 *                 that will then return the new current location.
+	 * Get the index in the string at the current moment.
```

Comment:
Suggestion: `Get the current index in the string`. The `current moment` feels a bit weird, like it can magically change when you look at it long enough :)